granulate_utils/metadata/databricks_client.py

```diff
@@ -85,24 +115,25 @@ def extract_relevant_metadata(self) -> Optional[Dict[str, str]]:
         start_time = time.monotonic()
         while time.monotonic() - start_time < DATABRICKS_JOBNAME_TIMEOUT_S:
             try:
-                if cluster_all_props := self._cluster_all_tags_metadata():
-                    self.logger.info(
-                        "Successfully got relevant cluster tags metadata",
-                        cluster_all_props=cluster_all_props,
-                    )
-                    return cluster_all_props
-                else:
-                    # No environment metadata yet, retry.
-                    time.sleep(RETRY_INTERVAL_S)
-            except DatabricksJobNameDiscoverException:
-                self.logger.exception("Failed to get DBX environment properties")
-                return None
-            except Exception:
-                self.logger.exception("Generic exception was raise during DBX environment properties discovery")
-                return None
+                return self._fetch_cluster_all_tags()
+
+            except Exception as e:
+                self.logger.exception("Exception was raised while getting DBX environment metadata")
+                if self.raise_on_failure:
+                    raise e
+
             if not self.enable_retries:
                 break
-        self.logger.info("Databricks get DBX environment metadata timeout, continuing...")
+
+            # No environment metadata yet, retry.
+            time.sleep(RETRY_INTERVAL_S)
+
+        self.logger.info("Databricks get DBX environment metadata timeout reached")
```

Comment:
This message looks redundant if we are raising the exception below. Maybe we should only log this if we do not raise?